vscode/microsoft-kiota/src/workspaceTreeProvider.ts

```diff
@@ -6,42 +6,54 @@ import { getWorkspaceJsonPath } from './util';
 
 
 export class WorkspaceTreeProvider implements vscode.TreeDataProvider<vscode.TreeItem> {
-    constructor(public isWSPresent: boolean) {
-    }
-    async getChildren(element?: vscode.TreeItem): Promise<vscode.TreeItem[]> {
-      if (!this.isWSPresent) {
-        return [];
-      }
-      if (!element) {
-        return [new vscode.TreeItem(KIOTA_WORKSPACE_FILE, vscode.TreeItemCollapsibleState.None)];
-      }
+  public isWSPresent: boolean;
+  private _onDidChangeTreeData: vscode.EventEmitter<vscode.TreeItem | undefined | null | void> = new vscode.EventEmitter<vscode.TreeItem | undefined | null | void>();
+  readonly onDidChangeTreeData: vscode.Event<vscode.TreeItem | undefined | null | void> = this._onDidChangeTreeData.event;
```

Comment:
I'm not sure I understand what's going on here.
Nothing subscribes to this event?

Reply:
Yeah, I don't have explicit listeners for this. I'm using it mainly to inform VS Code that data in the tree view has changed and the tree view needs to be refreshed. E.g when a generation occurs and a workspace.json is available

Reply:
Thank you for the additional information . 
Then if you don't have any subscribers, how does the refresh happen when you trigger the event?

Reply:
My understanding is that VS Code handles this internally... I guess there's a way it listens to the event
here's a reference to it
https://code.visualstudio.com/api/extension-guides/tree-view#updating-tree-view-content